### ISSUE TITLE
Extract the game behavior into its own class

### ISSUE THREAD

**Issue by jacob5567** [jacob5567/group-project-the-wet-bandits#1]
Currently, many of the game physics and mechanics are in the `Board` class, where they should not belong. The Board class, in its current form, performs far more functions than it should in order to be most reusable. In order for the Board class to not go beyond its purview of keeping track of the positions and state of the Board itself, I propose that a `Physics` or `Behavior` class be created to take on some of this load. The class would take on several of the existing methods into itself, such as `updatePieceLocations()`, `shuffleBoard()`, `inBounds()`, `removeMatches()`, and `removeChain()`. All of these functions are more concerned with the *behavior* of the game rather than the *state* of the game, and should thus be relocated. A general "move" function should also be implemented in order to facilitate the movement of pieces across the board.

**Comment by onzfonz** [comment]
Yeah so similar to #2, I would say to just add a proof of concept that would show that the behavior could be linked to a new state, or that another piece could take on slightly different behavior

**Comment by jacob5567** [comment]
Yeah, I could add a new proof-of-concept mode that would alter the game state in a new way, thus justifying the separation of the behavior and board.
